Tests/Recurly/Invoice_Test.php

```diff
@@ -121,4 +121,26 @@ public function testGetInvoicePdf() {
     $result = Recurly_Invoice::getInvoicePdf('1001', 'en-GB', $this->client);
     $this->assertEquals(array('/invoices/1001', 'en-GB'), $result);
   }
+
+  public function testRefundAmount() {
+    $this->client->addResponse('POST', 'https://api.recurly.com/v2/invoices/1001/refund', 'invoices/refund-201.xml');
+    $invoice = Recurly_Invoice::get('1001', $this->client);
+
+    $refund_invoice = $invoice->refundAmount(1000);
+    $this->assertEquals($refund_invoice->subtotal_in_cents, -1000);
+  }
+
+  public function testRefund() {
+    $this->client->addResponse('POST', 'https://api.recurly.com/v2/invoices/1001/refund', 'invoices/refund-201.xml');
+    $invoice = Recurly_Invoice::get('1001', $this->client);
+    $line_items = $invoice->line_items;
+
+    $adjustment_map = function($line_item) {
+      return array("uuid" => $line_item->uuid, "quantity" => $line_item->quantity, "prorate" => false);
+    };
+    $adjustments = array_map($adjustment_map, $line_items);
```

Comment:
This is the pattern we use for the other libs:

https://github.com/recurly/recurly-client-ruby/blob/master/spec/recurly/invoice_spec.rb#L54
https://github.com/recurly/recurly-client-python/blob/master/tests/test_resources.py#L573

I agree it's not ideal. What did you have in mind? Perhaps we have a method on the `Recurly_Adjustment` class? We can't just use the adjustment itself because it needs the `prorated` field and they may not want to refund the entire quantity.